src/mono/wasm/runtime/src/mono/scheduling.ts

```diff
@@ -0,0 +1,78 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import * as cwraps from './cwraps'
+
+const timeout_queue: Function[] = [];
+let spread_timers_maximum = 0;
+let isChromium = false;
+let pump_count = 0;
+
+if (globalThis.navigator) {
+    const nav: any = globalThis.navigator;
+    if (nav.userAgentData && nav.userAgentData.brands) {
+        isChromium = nav.userAgentData.brands.some((i: any) => i.brand == 'Chromium');
+    }
+    else if (nav.userAgent) {
+        isChromium = nav.userAgent.includes("Chrome");
+    }
+}
+
+function pump_message() {
```

Comment:
I think there are cases where we'll want to call this manually, so we may want to export it

Reply:
I believe that `schedule_background_exec` was/is exactly that.

Reply:
```schedule_background_exec``` queues a pump operation, you might want to synchronously call pump_message to implement something like ```Task.Wait```